docs/Agent/user-guide/man_page.md

```diff
@@ -1 +1,439 @@
+
+`pbench-register-tool-set [--toolset=<tool-set>] [--group=<group-name>] [--interval=<interval>] [--no-install] [--remotes=@<remotes-file>] [<tool-set>]`
+
+**DESCRIPTION**  
+Register all the tools in the specified toolset.
+
+**OPTIONS**  
+ --toolset  
+Available tool sets from /opt/pbench-agent/config/pbench-agent.cfg:
+
+- heavy
+- legacy
+- light
+- medium
+
+--remotes  
+Single remote host, a list of remote hosts (comma-separated, no spaces) or an "at" sign ("@") followed by a filename. In this last case, the file should contain a list of hosts and their (optional) labels. Each line of the file should contain a hostname, optionally followed by a label separated by a comma (","); empty lines are ignored, and comments are denoted by a leading hash, or pound ("#"), character.
+
+--labels  
+Where the list of labels must match the list of remotes.
+
+--help  
+Show this message and exit.
+
+---
+
+### pbench-register-tool-trigger
+
+---
+
+**NAME**  
+pbench-register-tool-trigger, Registers the tool trigger
+
+**SYNOPSIS**  
+`pbench-register-tool-trigger[-C][-g][--start-trigger][--stop-trigger][--help]`
+
+**DESCRIPTION**  
+Registers tool with the given group and start and stop tool trigger text
+
+**OPTIONS**  
+ -C, --config PATH  
+Path to a pbench-agent configuration file (defaults to the '\_PBENCH_AGENT_CONFIG' environment variable, if defined) [required]
+
+-g, --group,--groups STR  
+Registers the trigger in the <group-name> group. If no group is specified, the 'default'
+
+--start-trigger STR  
+[required]
+
+--stop-trigger STR  
+[required]
+
+--help  
+Show this message and exit.
+
+---
+
+### pbench-results-move
+
+---
+
+**NAME**  
+pbench-results-move, Move results tarball to the server
+
+**SYNOPSIS**  
+`pbench-results-move[-C][--controller][--token][--delete][--xz-single-threaded][--show-server][--help]
+`
+
+**DESCRIPTION**  
+Move result directories to the configured Pbench server.This command is used when pushing result tarball using token.
+
+**OPTIONS**  
+ -C, --config PATH  
+Path to a pbench-agent configuration file (defaults to the '\_PBENCH_AGENT_CONFIG' environment variable,if defined) [required]
+
+--controller STR  
+Override the default controller name
+
+--token STR  
+pbench server authentication token [required]
+
+--delete / --no-delete  
+Remove local data after successful copy [default: delete]
+
+--xz-single-threaded  
+Use single-threaded compression with 'xz'
+
+--show-server STR  
+Display information about the pbench server where the result(s) will be moved (Not implemented)
+
+--help  
+Show this message and exit.
+
+---
+
+### pbench-user-benchmark
+
+---
+
+**NAME**  
+pbench-user-benchmark, Control start/stop/post-process-tools.
+
+**SYNOPSIS**  
+`pbench-user-benchmark[-C][--tool-group][--iteration-list][--sysinfo][--pbench-pre][--pbench-post][--use-tool-triggers][--no-stderr-capture]-- <script to run>`
+
+**DESCRIPTION**  
+Collects data from the tools registered. Here are the steps involved
```

Comment:
Collects data from the registered tools while running a user-specified action, maybe. This can be a specific synthetic benchmark workload, a real workload, or simply a delay to measure system activity.